semanticdb/metacp/src/main/scala/scala/meta/internal/scalacp/SymbolInformationOps.scala

```diff
@@ -181,28 +181,33 @@ trait SymbolInformationOps { self: Scalacp =>
                   sym.infoType
                 }
               }
-              val stpe = tpe.toSemantic(linkMode)
+              val ssig = sig.toSemanticSig(linkMode)
               if (sym.isConstructor) {
-                stpe match {
-                  case t: s.MethodType => t.copy(returnType = s.NoType)
-                  case _ => stpe
+                ssig match {
+                  case ssig: s.MethodSignature => ssig.copy(returnType = s.NoType)
+                  case _ => ssig
                 }
               } else if (sym.isScalacField) {
-                val stparams = Some(s.Scope())
-                val sparamss = Nil
-                val sret = stpe
-                s.MethodType(stparams, sparamss, sret)
+                ssig match {
+                  case ssig: s.ValueSignature =>
+                    val stparams = Some(s.Scope())
+                    val sparamss = Nil
+                    val sret = ssig.tpe
+                    s.MethodSignature(stparams, sparamss, sret)
+                  case _ =>
+                    sys.error(s"unsupported signature: ${ssig.getClass} $ssig")
+                }
```

Comment:
This is another cute consequence of the Type/Signature split. We actually have to make sure that `ssig` is `ValueSignature` here. Otherwise, it won't fit into the return type of `MethodSignature`, which expects `s.Type`. It is pleasing to be more and more precise about our domain.